android/src/main/kotlin/cz/dronetag/flutter_opendroneid/scanner/ODIDScanner.kt

```diff
@@ -0,0 +1,69 @@
+package cz.dronetag.flutter_opendroneid
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+
+/// Contains common functinality for ODID scanners 
+/// Derived scanners should use receiveData method that takes raw data and metadata
+/// and sends it to stream
+/// Creates [ODIDPayload] instances implementing Pigeon PayloadAPI
+abstract class ODIDScanner(
+    private val odidPayloadStreamHandler: StreamHandler
+) : Pigeon.PayloadApi {
+
+    companion object {
+        const val MAX_MESSAGE_SIZE = 25
+        const val BT_OFFSET = 6
+        const val MAX_BLE_ADV_SIZE = 31
+        const val WIFI_BEACON_OFFSET = 5
+        const val WIFI_NAN_OFFSET = 1
```

Comment:
This might go to the `WifiNaNScanner` companion object instead? Same for BT if there's something.